internal/util/validation/validation.go

```diff
@@ -107,6 +113,71 @@ func ValidateBearerToken(token *string, path string) []error {
 	return asErrors(errs)
 }
 
+func ValidateCSRUsages(u *[]string) []error {
+	errs := field.ErrorList{}
+	requiredAllOf := map[string]struct{}{
+		"clientAuth": {},
+		"CA:false":   {},
+	}
+	notAllowed := map[string]struct{}{}
+
+	for _, usage := range *u {
+		if _, exists := notAllowed[usage]; exists {
+			err := fmt.Sprintf("usage not allowed: %s\n", usage)
+			errs = append(errs, field.Invalid(fieldPathFor("spec.usages"), u, err))
+		}
+		delete(requiredAllOf, usage)
+	}
+
+	l := len(requiredAllOf)
+	if l > 0 {
+		required := make([]string, l)
+		for k := range requiredAllOf {
+			required = append(required, k+", ")
+		}
+		err := fmt.Sprintf("required usages must be present in request: %s\n", required)
+		errs = append(errs, field.Invalid(fieldPathFor("spec.usages"), u, err))
+	}
+	return asErrors(errs)
+}
+
+// Currently every request is sent to the only signer, named "ca" and defined in cmd/flightctl-api/main.go
+func ValidateSignerName(s string) []error {
+	errs := field.ErrorList{}
+
+	validSigners := map[string]struct{}{
+		"ca": {},
+	}
+
+	if _, exists := validSigners[s]; exists {
+		return nil
+	}
+
+	errs = append(errs, field.Invalid(fieldPathFor("spec.signerName"), s, "must specify a valid signer"))
+	return asErrors(errs)
+}
+
+func ValidateExpirationSeconds(e *int32) []error {
+	if e != nil {
+		if *e < minExpirationSeconds {
+			log.Printf("WARNING: recommended minimum for ExpirationSeconds is %d", minExpirationSeconds)
```

Comment:
The user won't see this (it will go to the service's log), and it's also not using our configured logger.  I think you should turn this into a TODO comment and open a JIRA ticket for further handling.